src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1109,10 +1111,11 @@ void CompileBroker::possibly_add_compiler_threads(Thread* THREAD) {
 #endif
       JavaThread *ct = make_thread(compiler_t, compiler2_object(i), _c2_compile_queue, _compilers[1], THREAD);
       if (ct == NULL) break;
+      ThreadsListHandle tlh;  // get_thread_name() depends on the TLH.
+      assert(tlh.includes(ct), "ct=" INTPTR_FORMAT " exited unexpectedly.", p2i(ct));
```

Comment:
This can be placed inside the if-clause where the MutexLocker was.

Reply:
True. That would be better.